Sources/WebAuthn/WebAuthnError.swift

```diff
@@ -12,57 +12,120 @@
-    case invalidAuthData
-    case invalidFmt
-    case missingAttStmt
-    case attestationFormatNotSupported
+    public static let invalidAttestationObject = Self(reason: .invalidAttestationObject)
+    public static let invalidAuthData = Self(reason: .invalidAuthData)
+    public static let invalidFmt = Self(reason: .invalidFmt)
+    public static let missingAttStmt = Self(reason: .missingAttStmt)
+    public static let attestationFormatNotSupported = Self(reason: .attestationFormatNotSupported)
 
     // MARK: ParsedCredentialCreationResponse
-    case invalidCredentialCreationType
-    case credentialRawIDTooLong
+    public static let invalidCredentialCreationType = Self(reason: .invalidCredentialCreationType)
+    public static let credentialRawIDTooLong = Self(reason: .credentialRawIDTooLong)
 
     // MARK: AuthenticatorData
-    case authDataTooShort
-    case attestedCredentialFlagNotSet
-    case extensionDataMissing
-    case leftOverBytesInAuthenticatorData
-    case credentialIDTooLong
-    case credentialIDTooShort
+    public static let authDataTooShort = Self(reason: .authDataTooShort)
+    public static let attestedCredentialFlagNotSet = Self(reason: .attestedCredentialFlagNotSet)
+    public static let extensionDataMissing = Self(reason: .extensionDataMissing)
+    public static let leftOverBytesInAuthenticatorData = Self(reason: .leftOverBytesInAuthenticatorData)
+    public static let credentialIDTooLong = Self(reason: .credentialIDTooLong)
+    public static let credentialIDTooShort = Self(reason: .credentialIDTooShort)
 
     // MARK: CredentialPublicKey
-    case badPublicKeyBytes
-    case invalidKeyType
-    case invalidAlgorithm
-    case invalidCurve
-    case invalidXCoordinate
-    case invalidYCoordinate
-    case unsupportedCOSEAlgorithm
-    case unsupportedCOSEAlgorithmForEC2PublicKey
-    case invalidModulus
-    case invalidExponent
-    case unsupportedCOSEAlgorithmForRSAPublicKey
-    case unsupported
+    public static let badPublicKeyBytes = Self(reason: .badPublicKeyBytes)
+    public static let invalidKeyType = Self(reason: .invalidKeyType)
+    public static let invalidAlgorithm = Self(reason: .invalidAlgorithm)
+    public static let invalidCurve = Self(reason: .invalidCurve)
+    public static let invalidXCoordinate = Self(reason: .invalidXCoordinate)
+    public static let invalidYCoordinate = Self(reason: .invalidYCoordinate)
+    public static let unsupportedCOSEAlgorithm = Self(reason: .unsupportedCOSEAlgorithm)
+    public static let unsupportedCOSEAlgorithmForEC2PublicKey = Self(reason: .unsupportedCOSEAlgorithmForEC2PublicKey)
+    public static let invalidModulus = Self(reason: .invalidModulus)
+    public static let invalidExponent = Self(reason: .invalidExponent)
+    public static let unsupportedCOSEAlgorithmForRSAPublicKey = Self(reason: .unsupportedCOSEAlgorithmForRSAPublicKey)
+    public static let unsupported = Self(reason: .unsupported)
+}
+
+extension WebAuthnError {
```

Comment:
I'd prefer this to be in the main type rather than an extension given it's referenced from the main body

Reply:
Updated and moved it to the top.